### ISSUE TITLE
[Windows] [Vivaldi] Browser addon cannot connect to database

### ISSUE THREAD

**Issue by xanathon** [keepassxreboot/keepassxc#11407]
## Overview
Since installation of 2.7.9 the browser addon can no longer connect to database in Vivaldi (tried on three computers). Task Manager shows that proxy is running.

## Steps to Reproduce
Browser plugin does not work ("x" shown on icon)
opening plugin settings
clicking "connect database"
nothing happens

## Expected Behavior
browser plugin should work as before 2.7.9

## Actual Behavior
connection to database not possible, despite KeepassXC and proxy running

## Context
I uninstalled 2.7.9 and reinstalled 2.7.8 -> works immediately. Vivaldi is allowed in browser integration settings

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - 2.7.9
I cannot give you the info on detailed version or debug info, because I urgently needed the browser integration and so uninstalled and went back to 2.7.8

[NOTE]: # ( Pick choices based on your environment )
Operating System: Windows 10/64 latest patchlevel

**Comment by varjolintu** [comment]
If your database is already connected to Vivaldi, just by updating KeePassXC should not trigger the "Connect database" button to be visible.
You can try to delele all extension connections from your database (from Database Settings) and the keys from browser extension side (Connected Databases tab). Then restart everything just in case and try connecting your database again.

**Comment by droidmonkey** [comment]
No code was changed for browser connection between 2.7.8 and 2.7.9.

If you are using the portable version, make sure you put in the same directory or you'll have to apply the browser settings again.

**Comment by xanathon** [comment]
I do not use a portable version.

What I did in Vivaldi was deinstalling and reinstalling the KeepassXC extension, but that did not help.

In the "connected databases tab" no connection was active or even visible (there was no database entry there at all), so I could not remove any. The reconnect button in the browser extension did nothing.

I can only stress that 2.7.9 does not work, 2.7.8 works immediately and without any problems whatsoever.

**Comment by droidmonkey** [comment]
I think you just need to reinstall 2.7.9 with Vivaldi completely closed, and then maybe even reboot. It's likely the install got blocked by Vivaldi holding onto the proxy executable. A reboot should have cleared it, but best to update without Vivaldi running.

**Comment by xanathon** [comment]
I installed 2.7.9 yesterday evening and then shutdown the computers. When I started working again this morning I encountered the problem in the browsers.

Edit: KeepassXC is in autostart, but the browser is started manually (so way after KeepassXC).

**Comment by varjolintu** [comment]
Take a look at the Troubleshooting Guide and report if you find anything unusual:
https://github.com/keepassxreboot/keepassxc-browser/wiki/Troubleshooting-guide

**Comment by xanathon** [comment]
Thanks for the suggestion, but I already looked into that, that's why I checked if proxy is running. But everything else in the troubleshooting guide is quite extensive and time consuming. I will see if I find time for this, but at the moment to get it running fast and reliably my only reasonable solution for the time being is going back to 2.7.8.

Since that is working without any problems whatsoever and uses the same appdata and registry keys I doubt there is something wrong with those. The registry keys and appdata on one computer also does not explain why I have this problem on two desktop computers and two laptops.

**Comment by varjolintu** [comment]
It is a little time consuming, but it's the only thing to check if there's actually something strange happening. There should be no changes concerning how the extension connects to KeePassXC between 2.7.8 and 2.7.9.

**Comment by droidmonkey** [comment]
I installed Vivaldi and KeePassXC-Browser extension and it worked _immediately_ with 2.7.9. Unfortunately without reproduction we cannot fix anything.

**Comment by xanathon** [comment]
I would have expected you to **at least** give me some time to try the things in Troubleshooting mentioned above, especially as I mentioned that I will try to find time for it. I have to admit that I find it highly disrespectful just closing the issue. "Works here" never is an accepatble "solution". Especially as this happens on multiple computers at my side as I reported. And also especially flagging the issue "completed" is factually wrong. Just closing the issue does not make it magically vanish.

**Comment by varjolintu** [comment]
Let's keep this open.

**Comment by xanathon** [comment]
I got it working again.

Today the browser plugins suddenly stopped working with 2.7.8 also. My setup is not a standard user setup. I have a home network setup for my private computers laptops and for my work related computers and laptops. On the desktop computers I always have two instances of Vivaldi running, for multiple reasons.

My KeepassXC database is on a NAS, so all of the computers can access it and the database is the same for all of them.

Because of the latter the problem was present on all computers, at least that is my guess.

In the database settings I could see that there were multiple Browser entries (connections) for the same browser or browser instance. And this for all the browsers and browser instances in the local network. And that was why this happened on all computers in the local network: the shared database.

I deleted all browser connections and reconnected the browser plugins and for now it seems to work with 2.7.8 and 2.7.9.

For some reason sometimes KeepassXC does not seem to recognize the already connected browser plugin and then asks to reconfirm that browser via a user requester asking for a connection name. That adds another browser connection entry in the database settings for a browser that already exists in it (and my guess is that this leads to the encountered problem). Until now this never was a problem, at least I never encountered any problems with it. Until yesterday and accidentally that coincided with the installation of 2.7.9, but may actually be unrelated).

Since I do not know the code and how browser plugin connections are done, I have no idea how the multiple browser connection entries happen, but I think for UX it would be good to somehow double check if a browser connection already exists before asking the user to authorize the same browser again.

**Comment by varjolintu** [comment]
Double check that every browser with a different computer has a unique connection name with your database (e.g. vivaldi_work, vivaldi_home etc). Otherwise those will be overridden, or the name is found but the stored ID in the database does not match.

If you are using multiple databases, make sure you are not running a cloned databases (extension is using root UUIDs for identifying the database). To overcome this, it's possible to generate new root UUID for the database in the Database Settings.

> I think for UX it would be good to somehow double check if a browser connection already exists before asking the user to authorize the same browser again.

If the same connection name exists in the database, KeePassXC will ask if you want to override that with a new ID.

**Comment by droidmonkey** [comment]
@xanathon it's highly disrespectful to waste our time. You made no mention of this special setup in  your configuration. I had a hunch you had a very specific configuration at play and didn't tell us the moment it worked immediately. Respect the developers first by providing a complete report of your environment. 

We are happy to help, but swinging out the gates that we have a fault and not providing the whole story is not cool.

**Comment by xanathon** [comment]
> Double check that every browser with a different computer has a unique connection name with your database (e.g. vivaldi_work, vivaldi_home etc). Otherwise those will be overridden, or the name is found but the stored ID in the database does not match.

All my connections had unique names that identified each browser or browser instance clearly. However, when asked to create a new browser plugin connection, since the existing ones are not shown, the duplicate entries had different names.
 
> If you are using multiple databases, make sure you are not running a cloned databases (extension is using root UUIDs for identifying the database). To overcome this, it's possible to generate new root UUID for the database in the Database Settings.

I use only one database.

**Comment by xanathon** [comment]
> @xanathon it's highly disrespectful to waste our time. You made no mention of this special setup in your configuration. I had a hunch you had a very specific configuration at play and didn't tell us the moment it worked immediately. Respect the developers first by providing a complete report of your environment.
> 
> We are happy to help, but swinging out the gates that we have a fault and not providing the whole story is not cool.

First: In my opening post I mentioned a multiple computer setup. From what I see in the documentation using a database with multiple computers is not that exotic at all. Since I have no crystal ball and my telepathy is nearly nonexistent I do not know in detail what exact information you need. If you had a hunch you could have just asked me instead of now playing the blame game.

Aside from that: I **_immediately_** informed you here after I got it working, I started writing literally seconds after I found a solution. If you really think that is "disrespectful", your perception of "respect" seems to differ vastly from mine. But I will note that and next time I will use my time machine and prevent my past me from opening a ticket at all. ;) 

Plus: I could not know if that was a bug or not. And it now is established that there are problems with connecting to browser plugins and UX in that relation. I think that may be useful information, but what do I know ... ;)

**Comment by droidmonkey** [comment]
I think we're even and happy to carry on 🫡

**Comment by varjolintu** [comment]
> Plus: I could not know if that was a bug or not. And it now is established that there are problems with connecting to browser plugins and UX in that relation. I think that may be useful information, but what do I know ... ;)

I still doesn't know if it's a bug, a sync problem or something else. I'd need to get my hands on a similar live setup to see where the error actually happens, and what's the reason for strange behavior.

**Comment by xanathon** [comment]
Another update on this: Yesterday it happened again, and now I am no longer able to connect the vivaldi Addon at all, not with 2.7.8 and also not with 2.7.9. I reinstalled both and there are no longer duplicate entries for browsers in the database.

Since the plugin works without any problem whatsoever in Firefox, Chrome and Edge, I think that Vivaldi is the problem. A quick search in the Vivaldi community forums seems to confirm that, since this problems happened again and again in the past in vivaldi. It may be that a bug crept in again.

I'll try to install an older Vivaldi version when I find time and see if it works again. I'll then report here.

For others with the same problem: Until a solution is found you can use KeepassXCs autofill feature instead of the browser integration. It's not as comfortable, but it works.

**Comment by varjolintu** [comment]
That is the curse of Chromium forks. We have had similar reports for Brave too. Sometimes an update breaks the Native Messaging API, or something strange happens with the browser's child processes. Vivaldi's source code is not 100% open so I guess there's no way to check for relevant changes on their side.

Often this might be because of some internal ad-blocking feature that accidentally interferes with relevant APIs.

**Comment by xanathon** [comment]
I am now completely through the troubleshooting guide for the browser plugin, but it still does not work.

What I noticed: There are two KeepassXC entries in the Windows context menu.

<img width="148" alt="image" src="https://github.com/user-attachments/assets/9b588894-f9a7-4865-b550-4c3f4891ecf4">

**Comment by varjolintu** [comment]
Windows context menu? You mean Vivaldi's context menus?

**Comment by xanathon** [comment]
Sorry, yes, I meant Vivaldi's context menu. I now reinstalled KeepassXC and the browser plugin and still does not work. However, when I uninstalled the plugin from the browser plugin settings there still was a KeepassXC context menu entry. When I went to the Chrome plugin repository it still offered me to remove the extension, even if I already had removed it in the addon settings. After I clicked "remove" all context menu entries were gone. I then reinstalled everything, but still: nope. But now theree is only one context menu entry.

Worker window shows:

<img width="465" alt="image" src="https://github.com/user-attachments/assets/e1ab6740-a555-4f4b-b741-15bf8bd68964">
